### PR TITLE
Exclude action.func folders when processing prerenderFunctions

### PR DESCRIPTION
**Pull request by Adriel-M**
Fixes: #746 

Excludes this folder when processing `prerenderFunctions` since the `.vc-config.json` located in this folder contains `operationType == isr` and collected then validated later on. The validation looks for an associated `.prerender-config.json` in the root functions folder but doesn't have it for this new `.action.func` folder.

This folder was introduced in the 14.2.2 release specifically in this pr https://github.com/vercel/next.js/pull/64592

At the moment it doesn't have any functionality but might need to get revisited in the future.

### CODE REVIEW

**Comment by dario-piotrowicz** [comment]
@Adriel-M thanks so very much for this PR! ❤️

It looks great and works as intended 🙂 

However the code is inconsistent with the current logic we have in which we first collect all the functions and then ignore the invalid ones we can, this also results us in logging such ignored functions in `.vercel/output/static/_worker.js/nop-build-log.json` (which might or not be useful in this case 😅).

Also this PR's logic is being tested (thanks a lot for adding the test btw! ❤️) under `processPrerenderFunctions` although the logic isn't part of that function 😕.

I think we should ideally amend the above points before merging the PR, I've drafted here what I think the proper tweaks should be: https://github.com/cloudflare/next-on-pages/commit/bc60685e09f85280133118c619b07869b130b205

Please have a look and let me know if you'd be willing to make these changes 🙏
(if not please let me know and I can push the necessary changes to your branch and merge)

File: .changeset/funny-icons-marry.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@cloudflare/next-on-pages': patch
+---
+
+Exclude .action.func folder introduced in next.js 14.2.2
```
Comment (dario-piotrowicz) [inline .changeset/funny-icons-marry.md:5]:
```suggestion
Ignore the `.action.func` folders introduced in next.js 14.2.2
```

**Comment by Adriel-M** [comment]
@dario-piotrowicz cherry-picked your commit + populated the TODO comment. Let me know if it's good enough!

**Review by dario-piotrowicz — APPROVED** [review]

**Comment by dario-piotrowicz** [comment]
@Adriel-M all good (sorry I just updated the comment for consistency sake), thanks a lot for the PR again! ❤️ 🙏

**Pull request merged**